File: nodecore-miners-pop/src/main/java/nodecore/miners/pop/api/WebApiModule.java
// VeriBlock NodeCore
// Copyright 2017-2019 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.

package nodecore.miners.pop.api;

import com.google.inject.AbstractModule;
import com.google.inject.multibindings.Multibinder;

public class WebApiModule extends AbstractModule {
    @Override
    protected void configure() {
        Multibinder<ApiController> controllers = Multibinder.newSetBinder(binder(), ApiController.class);
        controllers.addBinding().to(MiningController.class);
        controllers.addBinding().to(ConfigurationController.class);
        controllers.addBinding().to(QuitController.class);
        controllers.addBinding().to(LastBitcoinBlockController.class);
    }
}

File: nodecore-miners-pop/src/main/java/nodecore/miners/pop/api/WebApiModule.kt
package nodecore.miners.pop.api

import nodecore.miners.pop.api.controller.ConfigurationController
import nodecore.miners.pop.api.controller.LastBitcoinBlockController
import nodecore.miners.pop.api.controller.MiningController
import nodecore.miners.pop.api.controller.QuitController
import nodecore.miners.pop.api.controller.WalletController
import org.koin.dsl.module

@JvmField
val webApiModule = module {
    single { ConfigurationController(get()) }
    single { MiningController(get()) }
    single { WalletController(get()) }
    single { LastBitcoinBlockController(get()) }
    single { QuitController() }

    single {
        ApiServer(listOf(
            get<ConfigurationController>(),
            get<MiningController>(),
            get<WalletController>(),
            get<LastBitcoinBlockController>(),
            get<QuitController>()
        ))
    }
}
